src/Uno.Extensions.Reactive/Core/State.Extensions.cs

```diff
@@ -135,7 +188,33 @@ public static ValueTask Set(this IState<string> state, string? value, Cancellati
 	/// <returns>A <see cref="IDisposable"/> that can be used to remove the callback registration.</returns>
 	public static IDisposable ForEachAsync<T>(this IState<T> state, AsyncAction<T?> action, [CallerMemberName] string? caller = null, [CallerLineNumber] int line = -1)
 		where T : notnull
-		=> new StateForEach<T>(state, action, $"ForEachAsync defined in {caller} at line {line}.");
+		=> new StateForEach<T>(state, action.SomeOrNone(), $"ForEachAsync defined in {caller} at line {line}.");
+
+	/// <summary>
+	/// Execute an async callback each time the state is being updated.
+	/// </summary>
+	/// <typeparam name="T">The type of the state</typeparam>
+	/// <param name="state">The state to listen.</param>
+	/// <param name="action">The callback to invoke on each update of the state.</param>
+	/// <param name="caller"> For debug purposes, the name of this subscription. DO NOT provide anything here, let the compiler fulfill this.</param>
+	/// <param name="line">For debug purposes, the name of this subscription. DO NOT provide anything here, let the compiler fulfill this.</param>
+	/// <returns>A <see cref="IDisposable"/> that can be used to remove the callback registration.</returns>
+	public static IDisposable ForEachAsync<T>(this IState<T?> state, AsyncAction<T?> action, [CallerMemberName] string? caller = null, [CallerLineNumber] int line = -1)
```

Comment:
Is this the new scenario that we're adding?

Reply:
Yes and no : we only add overloads on existing operators constrained on `T : notnull` to accept also `T : struct`. Will add details in the doc before next release.